pkg/store/cache/memcached.go

```diff
@@ -33,15 +34,9 @@ type RemoteIndexCache struct {
 	compressionScheme string
 
 	// Metrics.
-	postingRequests              prometheus.Counter
-	seriesRequests               prometheus.Counter
-	expandedPostingRequests      prometheus.Counter
-	postingHits                  prometheus.Counter
-	seriesHits                   prometheus.Counter
-	expandedPostingHits          prometheus.Counter
-	postingDataSizeBytes         prometheus.Observer
-	expandedPostingDataSizeBytes prometheus.Observer
-	seriesDataSizeBytes          prometheus.Observer
+	requestTotal  *prometheus.CounterVec
+	hitsTotal     *prometheus.CounterVec
+	dataSizeBytes *prometheus.HistogramVec
```

Comment:
I think there were some things removed accidentally here?

Reply:
All of those were defined for convenience and initialized at init time. For example:

`postingRequests = commonMetrics.requestTotal.WithLabelValues(cacheTypePostings)` . Prior to adding the tenant label, that made sense, as one would simply need to do i.e `postingRequests.Add(...` when interacting with this value, as the label was "static". 

However, now that we're adding a tenant label to all of these metrics, and as we don't know the tenant at init time, we need to type out the full thing every time we want to interact with this metric. So the above example has become:

`requestTotal.WithLabelValues(cacheTypePostings, tenant).Add(...`

Hopefully that makes sense.

Reply:
I mean the removal of all the other counter above these, like `postingRequests`, `seriesRequests`, etc. Not sure if it's intended or accidental.

Reply:
We're not removing anything here, just changing the way we reference these objects.

Before this change, we mapped all the metrics like this:

```
c.postingRequests = commonMetrics.requestTotal.WithLabelValues(cacheTypePostings)
c.seriesRequests = commonMetrics.requestTotal.WithLabelValues(cacheTypeSeries)
c.expandedPostingRequests = commonMetrics.requestTotal.WithLabelValues(cacheTypeExpandedPostings)

c.postingHits = commonMetrics.hitsTotal.WithLabelValues(cacheTypePostings)
c.seriesHits = commonMetrics.hitsTotal.WithLabelValues(cacheTypeSeries)
c.expandedPostingHits = commonMetrics.hitsTotal.WithLabelValues(cacheTypeExpandedPostings)

c.postingDataSizeBytes = commonMetrics.dataSizeBytes.WithLabelValues(cacheTypePostings)
c.seriesDataSizeBytes = commonMetrics.dataSizeBytes.WithLabelValues(cacheTypeSeries)
c.expandedPostingDataSizeBytes = commonMetrics.dataSizeBytes.WithLabelValues(cacheTypeExpandedPostings)

c.postingsFetchDuration = commonMetrics.fetchLatency.WithLabelValues(cacheTypePostings)
c.seriesFetchDuration = commonMetrics.fetchLatency.WithLabelValues(cacheTypeSeries)
c.expandedPostingsFetchDuration = commonMetrics.fetchLatency.WithLabelValues(cacheTypeExpandedPostings)
```

I.e there are 4 different metrics (3 before recent rebase),  `requestTotal`,  `hitsTotal`, `dataSizeBytes`, `fetchLatency`, each of these were initialized with 3 labels, resulting in 4x3 counters/observers, as seen in the mapping above.

With this PR, because we're introducing a "dynamic" label (i.e the tenant) we have to work of the base `CounterVec`/`HistrogramVec` objects, when we interact with these metrics, otherwise we cannot add the tenant label. So I removed the convenience mapping, and put the base objects there instead.